packages/odyssey-react-mui/src/labs/SideNav.tsx

```diff
@@ -577,16 +589,43 @@ const SideNav = ({
 }: SideNavProps) => {
   const [isSideNavCollapsed, setSideNavCollapsed] = useState(false);
   const odysseyDesignTokens = useOdysseyDesignTokens();
+  const [sideNavItemIdToScrollTo, setSideNavItemIdToScrollTo] =
+    useState<string>("");
+  const scrollToNode = useCallback<RefCallback<HTMLLIElement>>((node) => {
+    node?.scrollIntoView({
+      behavior: "instant",
+      block: "end",
+      inline: "nearest",
+    });
+  }, []);
+  const scrollToNodeRefIfApplicable = useCallback(
+    (itemId: string) =>
+      itemId === sideNavItemIdToScrollTo ? scrollToNode : undefined,
+    [scrollToNode, sideNavItemIdToScrollTo],
+  );
 
   const processedSideNavItems = useMemo(
     () =>
       sideNavItems.map((item) => ({
         ...item,
-        children: item.children?.map((childProps) => (
-          <SideNavItemContent key={childProps.id} {...childProps} />
-        )),
+        children: item.children?.map((childProps) => {
+          if (
+            childProps.isSelected &&
+            (item.isExpanded || item.isDefaultExpanded) &&
+            !sideNavItemIdToScrollTo
+          ) {
+            setSideNavItemIdToScrollTo(childProps.id);
+          }
+          return (
+            <SideNavItemContent
+              key={childProps.id}
+              ref={scrollToNodeRefIfApplicable(childProps.id)}
+              {...childProps}
+            />
+          );
+        }),
       })),
-    [sideNavItems],
+    [scrollToNodeRefIfApplicable, sideNavItemIdToScrollTo, sideNavItems],
   );
```

Comment:
`setSideNavItemIdToScrollTo(childProps.id);` is a side-effect.

Side-effects should go in `useEffect`. No one would expect them in `useMemo`, and that's now how it's meant to be used. We're essentially overloading the way it works by doing that.